drivers/net/wireless/mediatek/mt76/mt76_connac.h

```diff
@@ -134,12 +134,16 @@ mt76_connac_pm_unref(struct mt76_connac_pm *pm)
 static inline bool
 mt76_connac_skip_fw_pmctrl(struct mt76_phy *phy, struct mt76_connac_pm *pm)
 {
+	struct mt76_dev *dev = phy->dev;
 	bool ret;
 
 	spin_lock_bh(&pm->wake.lock);
 	ret = pm->wake.count || test_and_set_bit(MT76_STATE_PM, &phy->state);
 	spin_unlock_bh(&pm->wake.lock);
 
+	if (mt76_is_mmio(dev) && dev->token_count)
+		return true;
+
```

Comment:
@objelf I guess you need to check token_count before setting MT76_STATE_PM otherwise we will return true (so not performing fw_own) and state is marked in PM state. Another approach would be to clear MT76_STATE_PM before returning true

Reply:
ack i will check about it.